### DISCUSSION TITLE
Jitters with axis merging

### DISCUSSION THREAD

**Discussion by boris27L** [WhiteMagic/JoystickGremlin#577] [Q&A]
Hi, 

I run a flight sim cockpit with dual inputs (dual yokes, dual pedals). I can connect all devices to the sim (having duplicate assignments for the affected axes) but that's not ideal.

JSG appears to be a great tool to solve this problem. My understanding is that merging axes means taking the sum of the physical inputs and sending it to the virtual axis. Means when one is steering fully to the right and the other one fully to the left, the output would be zero.

I'm not sure, however, what the merge operation does. "Sum" gives me only 50% output of the respective input, "Average" is always zero, only "minimum" seems to do the trick.

But I get massive spikes and jitters on the output, which makes it unusable. I have already installed a plugin called deadzone_filter, but that seems to do nothing. Did I do something wrong or ist it just as it is? The input devices are all BU836 controllers.

Thie is how it looks like. Displayed below is one of the physical devices.

![image](https://github.com/user-attachments/assets/f317cf3f-5217-4a8f-93ad-b89e03e681cb)

Here is my merge:
![image](https://github.com/user-attachments/assets/20d880e2-b423-4e50-bb77-85b9d6ffa31a)

And this is how I assigned the deadzone plugin:
![image](https://github.com/user-attachments/assets/155758a5-655b-4ce7-93f8-61a0d7b9de8d)
![image](https://github.com/user-attachments/assets/ba5b3bc3-4dcc-49b3-a3a3-3f005e395daa)

Regards
Boris

**Comment by WhiteMagic** [comment]
The actual implementation of the merge axis functions can be found here: https://github.com/WhiteMagic/JoystickGremlin/blob/a8bb3c95e1df5687d0706ebaf059c86edd9ec581/gremlin/code_runner.py#L366

Looking at your merge axis setup, what do you expect to happen? You merge the same axis, so identical values, I'm not sure what that is supposed to achieve. Also if you're running both the merge axis and the deadzone plugin together then you will end up having both run and send vJoy events. In Gremlin everything reacts to events and the outcome of handling an event is not made available to any other handlers.

In any case when you have two axes that both should control the same vJoy output axis then the simplest case is a remap on both of them to the same output. However, that can easily cause values being reset or jitter if your physical axes have the tiniest amount of noise or jitter to them. If you want two axis to jointly control the full range the sum merge action might work, but depending on how you really want this to behave you may need a custom plugin.

**Comment by boris27L** [comment]
Thanks for looking into this :)

I did not merge the same axes. BU0836-2_X-axis should be merged with BU0836-3_Z-axis to vJoy_device_1_X-axis. That is what I saw in a YT video, where it worked fine. 

I have the same result without the deadzone plugin, which was said to help with jitters (found that one on [Reddit](https://www.reddit.com/r/hotas/comments/st26id/fixing_spikingghostingnoisyfluctuating_axis/?rdt=33725)).

A remap of both axes to the same output was even worse, as you described.

What I want is having two controllers for the same axis (e.g. two sets of pedals for the rudder axis in a flight simulator) without spikes, resets or jitter: The combined output of two inputs.

Would using different modes work? 
Mode 1: BU0836-2_X-axis to vJoy_device_1_X-axis
Mode 2: BU0836-3_Z-axis to vJoy_device_1_X-axis
Assigning a button to switch between the modes? Is that possible?
